common/tools/dev-tool/src/commands/run/testNodeJSInput.ts

```diff
@@ -17,14 +17,20 @@ export const commandInfo = makeCommandInfo(
       default: false,
       description: "whether to run with test-proxy",
     },
+    "no-esm": {
+      shortName: "nesm",
+      kind: "boolean",
+      default: false,
+      description: "whether to skip loading the esm package",
+    },
   },
 );
 
 export default leafCommand(commandInfo, async (options) => {
   const reporterArgs =
     "--reporter ../../../common/tools/mocha-multi-reporter.js --reporter-option output=test-results.xml";
   const defaultMochaArgs = `${
-    (await isModuleProject())
+    options["no-esm"] === true || (await isModuleProject())
```

Comment:
I was thinking to use an option to toggle between the two workarounds, for example with `--no-esm=true` and if it is not module project, we would have `"--loader=../../../esm4mocha.mjs -r source-map-support/register"` so we don't have to pass the loader option in the package's NPM script.